windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas

```diff
@@ -418,6 +427,102 @@ procedure TMergeKeyboardInfo.CheckOrAddID;
     json.AddPair('id', FID);
 end;
 
+procedure TMergeKeyboardInfo.AddSubtagNames(id: String; o: TJSONObject);
+var
+  displayName, languageName, scriptName, regionName: String;
+  v: TJSONValue;
+  bcp47Tag: TBCP47Tag;
+begin
+  if id = '' then
+    Exit;
+  bcp47Tag := TBCP47Tag.Create(id);
+  try
+    TLanguageCodeUtils.BCP47Languages.TryGetValue(bcp47Tag.Language, languageName);
+    TLanguageCodeUtils.BCP47Scripts.TryGetValue(bcp47Tag.Script, scriptName);
+    TLanguageCodeUtils.BCP47Regions.TryGetValue(bcp47Tag.Region, regionName);
+  finally
+    bcp47Tag.Free;
+  end;
+
+  displayName := TLanguageCodeUtils.LanguageName(languageName, scriptName, regionName);
+
+  v := o.Values[TKeyboardInfoFile.SDisplayName];
+  if not Assigned(v) then
+    o.AddPair(TKeyboardInfoFile.SDisplayName, displayName);
+
+  v := o.Values[TKeyboardInfoFile.SLanguageName];
+  if not Assigned(v) then
+    o.AddPair(TKeyboardInfoFile.SLanguageName, languageName);
+
+  if scriptName <> '' then
+  begin
+    v := o.Values[TKeyboardInfoFile.SScriptName];
+    if not Assigned(v) and (scriptName <> '') then
+      o.AddPair(TKeyboardInfoFile.SScriptName, scriptName);
+  end;
+
+  if regionName <> '' then
+  begin
+    v := o.Values[TKeyboardInfoFile.SRegionName];
+    if not Assigned(v) and (regionName <> '') then
+      o.AddPair(TKeyboardInfoFile.SRegionName, regionName);
+  end;
+end;
+
+procedure TMergeKeyboardInfo.CheckOrMigrateLanguages;
+var
+  v: TJSONValue;
+  alangs: TJSONArray;
+  olangs, o: TJSONObject;
+  pair: TJSONPair; 
+  i: Integer;
+  id: string;
+begin
+  v := json.GetValue(TKeyboardInfoFile.SLanguages);
+
+  if v is TJSONArray then
+  begin
+    // Migrate languages[] array to Object
+    alangs := v as TJSONArray;
+    olangs := TJSONObject.Create;
+    try
+      for i := 0 to alangs.Count - 1 do
+      begin
+        id := alangs.Items[i].Value;
+        if id = '' then
+          continue;
+
+        // Populate subtag names  
+        o := TJSONObject.Create;
+        olangs.AddPair(id, o);
+        AddSubtagNames(id, o);
+      end;
+
+      json.RemovePair(TKeyboardInfoFile.SLanguages);
+      json.AddPair(TKeyboardInfoFile.SLanguages, olangs);
+    finally
```

Comment:
Remove the `try`/`finally` altogether seeing as there is nothing in the `finally` clause.